### ISSUE TITLE
v0.6.0 scale-set breaks the ability to have 2 runner install with the same name

### ISSUE THREAD

**Issue by ba-work** [actions/actions-runner-controller#2905]
### Checks

- [X] I've already read https://github.com/actions/actions-runner-controller/blob/master/TROUBLESHOOTING.md and I'm sure my issue is not covered in the troubleshooting guide.
- [X] I'm not using a custom entrypoint in my runner image

### Controller Version

v0.6.0

### Helm Chart Version

v0.6.0

### Deployment Method

Helm

### cert-manager installation

no cert manager

### Checks

- [X] This isn't a question or user support case (For Q&A and community support, go to [Discussions](https://github.com/actions/actions-runner-controller/discussions). It might also be a good idea to contract with any of contributors and maintainers if your business is so critical and therefore you need priority support
- [X] I've read [releasenotes](https://github.com/actions/actions-runner-controller/tree/master/docs/releasenotes) before submitting this issue and I'm sure it's not due to any recently-introduced backward-incompatible changes
- [X] My actions-runner-controller version (v0.x.y) does support the feature
- [X] I've already upgraded ARC (including the CRDs, see charts/actions-runner-controller/docs/UPGRADING.md for details) to the latest and it didn't fix the issue
- [X] I've migrated to the workflow job webhook event (if you using webhook driven scaling)

### Resource Definitions
I _use to_ install the runners with flux like this:
```yaml
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: repo-1
  namespace: arc-runners
spec:
  values:
    runnerScaleSetName: ci-cluster # <-- note that this has the same value in repo-2
    ...
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: repo-2
  namespace: arc-runners
spec:
  values:
    runnerScaleSetName: ci-cluster # <-- note that this has the same value in repo-1
    ...
```

### To Reproduce

```markdown
helm install repo-1 \
    oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set \
    --set runnerScaleSetName=ci-cluster
helm install repo-2 \
    oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set \
    --set runnerScaleSetName=ci-cluster
```

### Describe the bug

I _use to_ install the runners with flux like the included `HelmRelease`s suggest; they could have the same value for `runnerScaleSetName` and workflows **in different repos** could both target `ci-cluster`. With v0.6.0 (this commit: https://github.com/actions/actions-runner-controller/commit/837a1cb8503e81d8330c87df9a1e9ba616753811) the resources overlap and it doesn't work with no ability to override/workaround. 

### Describe the expected behavior

The runner groups/scalesets are created with the same name in each repo, like they were in v0.5.0

### Whole Controller Logs

```shell
this isn't relevant, the bug is related to helm templating
```

### Whole Runner Pod Logs

```shell
this isn't relevant, the bug is related to helm templating
```

### Additional Context

had to revert to 0.5.0 since this change breaking all workflows in `repo-2` that expect to be able to target `ci-cluster`

**Comment by ba-work** [comment]
some more info, this works on 0.5.0 because both `AutoscalingRunnerSet`s have different names, but have the same annotation: `runner-scale-set-name=ci-cluster`

**Comment by Link-** [comment]
@ba-work - this worked previously because of a bug in our charts which was fixed in 0.6.0. The isolation model we designed for is:

1. Runner scale set name is unique per runner group (you can have multiple runner scale sets with the same name, if they belong to different runner groups)
2. Runner scale set name is unique per namespace

You were never supposed to be able to create 2 runner scale sets with the same name in the same namespace. Even if they were registered against different repositories, organisations or enterprises.

You can read more about this here: https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/deploying-runner-scale-sets-with-actions-runner-controller#using-arc-across-organizations

**Comment by Link-** [comment]
I'm closing this because there's not much we can do about this here. Feel free to migrate this to a discussion and we can continue the conversation there if need be.
